From: Ashok Ramakrishnan <ashok.ramakrishnan@simplivity.com>
Date: Tue, 16 Aug 2016 13:31:54 +0000
Subject: HYP-2212 : Ignore the fact that there is no share mode entry to
 reconnect to.

HyperV seems to want to reconnect a Persistent Handle without a share mode entry.
---
 source3/smbd/durable.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/source3/smbd/durable.c b/source3/smbd/durable.c
index 3be3734..e74d85b 100644
--- a/source3/smbd/durable.c
+++ b/source3/smbd/durable.c
@@ -682,9 +682,8 @@ NTSTATUS vfs_default_durable_reconnect(struct connection_struct *conn,
 	        break;
         }
         if ( e == NULL ) {
-                DEBUG(1, ("vfs_default_durable_reconnect: denying durable reconnect as no share mode entries available to reconnect\n"));
-                TALLOC_FREE(lck);
-                return NT_STATUS_OBJECT_NAME_NOT_FOUND;
+                DEBUG(1, ("vfs_default_durable_reconnect: Error : no share mode entries available to reconnect, Ignoring...\n"));
+                goto PROCEEDOPEN;
         }
 
 	if ((e->access_mask & (FILE_WRITE_DATA|FILE_APPEND_DATA)) &&
